### PR TITLE
Impl withdrawal function

### PR DESCRIPTION
**Pull request by Pvsaint**
Implemented and tested withdrawal function, implemented ERC20 token transfer and interface.
Set up a new ERC20 interface.

### CODE REVIEW

File: src/campaign.cairo [hunk 1]
```diff
@@ -116,20 +151,58 @@ mod TokengiverCampaign {
...
+
+            assert(caller == campaign.campaign_owner, NOT_CAMPAIGN_OWNER);
+
+            let available_balance: u256 = self.withdrawal_balance.read(campaign_address);
+            assert(amount <= available_balance, INSUFFICIENT_BALANCE);
+
+            self.withdrawal_balance.write(campaign_address, available_balance - amount);
+
+            // Transfer tokens
+            self.token.read().transfer_from(campaign_address, caller, amount);
```
Comment (mubarak23) [inline src/campaign.cairo:166]: please use IECR20DISPATCHER to handle this transfer,

look at the work on donate function to see how it was implemented

File: src/campaign.cairo [hunk 2]
```diff
@@ -116,20 +151,58 @@ mod TokengiverCampaign {
...
+            self.emit(Withdrawal { campaign_address, recipient: caller, amount });
+        }
+
+        fn donate(
```
Comment (mubarak23) [inline src/campaign.cairo:170]: why are you working on this function,

your issue is base on withdrawal, so please remove this work, someone already work on this function and the PR is merged.

File: src/interfaces/ICampaign.cairo [hunk 3]
```diff
@@ -20,6 +20,13 @@ pub trait ICampaign<TState> {
     fn set_donation_count(ref self: TState, campaign_address: ContractAddress);
     fn set_available_withdrawal(ref self: TState, campaign_address: ContractAddress, amount: u256);
     fn set_donations(ref self: TState, campaign_address: ContractAddress, amount: u256);
+    fn withdraw(ref self: TState, campaign_address: ContractAddress, amount: u256);
+    fn donate(
```
Comment (mubarak23) [inline src/interfaces/ICampaign.cairo:24]: remove this also

**Review by mubarak23 — CHANGES_REQUESTED** [review]
i will suggest you sync your fork and continue working from it

File: src/interfaces/ICampaign.cairo (hunk 3 above)
Comment (Pvsaint) [reply src/interfaces/ICampaign.cairo:24]: All reviews have been completed sir

**Comment by mubarak23** [comment]
fix your merge conflict please

**Comment by Pvsaint** [comment]
> fix your merge conflict please

Processing update, recent push is still being processed, and will show up in a bit.

**Comment by mubarak23** [comment]
> > fix your merge conflict please
> 
> Processing update, recent push is still being processed, and will show up in a bit.

Hmmm Ok

File: src/interfaces/ICampaign.cairo (hunk 3 above)
Comment (Pvsaint) [reply src/interfaces/ICampaign.cairo:24]: It is taking too long to update the push.

**Pull request closed without merging**